react/IOMessageWithMarkers.tsx

```diff
@@ -0,0 +1,36 @@
+import React, { useMemo } from 'react'
+import { useCssHandles } from 'vtex.css-handles'
+
+import IOMessage from './IOMessage'
+import { IOMessageWithMarkers as IOMessageWithMarkersType } from './typings/IOMessageWithMarkers'
+
+const IOMessageWithMarkers: IOMessageWithMarkersType = ({
+  label,
+  markers = [],
+  componentName,
+  values = {},
+}) => {
+  const CSS_HANDLES = useMemo(() => markers.map(marker => {
+    return `${componentName}_${marker}`
+  }), [markers])
+  const handles = useCssHandles(CSS_HANDLES)
+  
+  const markerComponents = markers.reduce((acc: any, marker) => {
+    acc[marker] = (...chunks: any) => (
+      <span className={handles[`${componentName}_${marker}`]}>{chunks}</span>
+    )
+    return acc
+  }, {})
```

Comment:
The type for `acc` and `markerComponents` could be `Record<string, ReactElement>`.
Also, I don't quite understand what is `...chunks`

Reply:
chunks[ it's something from react-intl](https://github.com/formatjs/react-intl/blob/master/docs/Components.md#rich-text-formatting)

![image](https://user-images.githubusercontent.com/284515/75570008-0745c180-5a35-11ea-84d4-9ade8219488e.png)